test/test_linalg.py

```diff
@@ -7453,6 +7454,54 @@ def test_preferred_linalg_library(self):
         self.assertEqual(out_ref, out1.cpu())
         self.assertEqual(out1, out2)
 
+    @onlyCUDA
+    @toleranceOverride({torch.float16: tol(atol=1e-1, rtol=1e-1),
+                        torch.bfloat16: tol(atol=1e-1, rtol=1e-1),
+                        torch.float32: tol(atol=1e-1, rtol=1e-1)})
+    @dtypes(torch.float16, torch.bfloat16, torch.float32)
+    @parametrize("size", [100, 1000, 10000])
+    def test_cublas_addmm(self, size: int, dtype: torch.dtype, device):
+        #
+        # Check for catastrophic cuBLAS inaccuracy by measuring the deviation between
+        # results from the CUDA invocation of torch.addmm and the CPU invocation
+        # (which does not use CUDA backend).
+        #
+        # Get dims
+        n, m, p = (size + 1, size, size + 2)
+        # Make random tensors on CPU
+        torch.manual_seed(123456789)
+        m_beta = make_tensor(1, dtype=dtype, device='cpu')
+        m_input = make_tensor((n, p), dtype=dtype, device='cpu')
+        m_1 = make_tensor((n, m), dtype=dtype, device='cpu')
+        m_2 = make_tensor((m, p), dtype=dtype, device='cpu')
+        # *FLOAT16 Special Handling*
+        # Backend does not tensorize float16 on CPU, so convert to float32
+        if dtype == torch.float16:
+            m_beta = m_beta.to(dtype=torch.float32)
+            m_input = m_input.to(dtype=torch.float32)
+            m_1 = m_1.to(dtype=torch.float32)
+            m_2 = m_2.to(dtype=torch.float32)
+        # Get CPU result
+        res_cpu = torch.addmm(m_input, m_1, m_2, beta=m_beta.item())
+        # *FLOAT16 Special Handling*
+        # Convert back to float16
+        if dtype == torch.float16:
+            m_beta = m_beta.to(dtype=torch.float16)
+            m_input = m_input.to(dtype=torch.float16)
+            m_1 = m_1.to(dtype=torch.float16)
+            m_2 = m_2.to(dtype=torch.float16)
+            res_cpu = res_cpu.to(dtype=torch.float16)
+        # Move tensors to GPU
+        m_beta = m_beta.to("cuda")
```

Comment:
`.to(device)`, and similarly in all other places

Reply:
Since this test is specific to cuBLAS, presumably it should only run for a CUDA device; removing "device" param